libs/vkd3d/swapchain.c

```diff
@@ -2035,6 +2188,12 @@ static void dxgi_vk_swap_chain_present_iteration(struct dxgi_vk_swap_chain *chai
     if (!chain->present.vk_swapchain)
         return;
 
+    /* If low latency is enabled we should only try to present once to avoid having to
+     * increment the present id for each failed present. Make sure to do this after checking
+     * if the swapchain needs to be recreated so the low latency state is up to date. */
+    max_present_iterations = chain->present.low_latency_state.mode ?
```

Comment:
This is pretty confusing. Won't this cause some presents to simply be dropped entirely? If OUT_OF_DATE is observed, the present won't go through, and it seems like the entire frame is dropped, so a particular presentID won't be observed by driver. Is this a problem that could cause deadlock?

Reply:
I just ended up deleting this, the present_id_valid logic should prevent the case this was trying to prevent.